File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/RemoveFileAttachmentObserverUseCase.java
package com.glia.widgets.core.fileupload.domain;

import com.glia.widgets.core.fileupload.FileAttachmentRepository;

import java.util.Observer;

public class RemoveFileAttachmentObserverUseCase {
    private final FileAttachmentRepository repository;

    public RemoveFileAttachmentObserverUseCase(FileAttachmentRepository repository) {
        this.repository = repository;
    }

    public void execute(Observer observer) {
        this.repository.removeObserver(observer);
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/fileupload/domain/RemoveFileAttachmentObserverUseCase.kt
package com.glia.widgets.core.fileupload.domain

import com.glia.widgets.core.fileupload.EngagementFileAttachmentRepository
import java.util.Observer

@Deprecated("This class is not needed anymore")
internal class RemoveFileAttachmentObserverUseCase(private val repository: EngagementFileAttachmentRepository) {
    operator fun invoke(observer: Observer?) {
        repository.removeObserver(observer)
    }
}
